TeamCharles/workforce-management
# 57-employee-edit

## Status
**READY**

## Documentation

[X] I have fully documented features in this PR's code. Please put an `X` in the box to confirm.

[ ] This PR does **NOT** require documentation.

## Description
Provide a form, pre-populated with existing Employee data, for the user to edit. On submit, update the Employee record and any Attendee records as necessary.

## Resolves Issue Number
Resolves #57 

## Deploy Notes
Might need to redo migrations on this one.

```sh
git pull --prune
git checkout 57-employee-edit
dotnet run
```

## Steps to Test or Reproduce

1. `localhost:5000/employee/edit/10`
1. Remove some required fields and try to submit (submit button should be disabled)
1. Try to assign a new computer
1. Assign/Unassign some Training Programs
1. Submit the form with changes
1. Verify that the Employee record was updated
1. Verify that any Attendee records were updated appropriately (added/removed)

## Impacted Files in Application
List general components of the application that this PR will affect:

* Views/Employee/Edit
* Views/Employee/Add
* Controllers/EmployeeController
* ViewModels/Employee/EmployeeForm
* Models/Employee
